### PR TITLE
core: bump rust to 1.76.0 + update

### PR DESCRIPTION
**Pull request by spolu**
## Description

https://github.com/dust-tt/dust/security/dependabot/76

Security vuln in dependency requires an update. Updates require bumping rust to 1.73.0

cc @dust-tt/engineering-team make sure to
```
rustup default 1.76.0
```

## Risk

Minimal

## Deploy Plan

- deploy `core`

### CODE REVIEW

File: core/src/data_sources/data_source.rs [hunk 1]
```diff
@@ -638,6 +638,7 @@ impl DataSource {
                 match qdrant_client
                     .set_payload(
                         self.qdrant_collection().to_string(),
+                        None,
```
Comment (fontanierh) [inline core/src/data_sources/data_source.rs:641]: These changes are unrelated right ?

**Review by fontanierh — APPROVED** [review]

File: core/src/data_sources/data_source.rs (hunk 1 above)
Comment (spolu) [reply core/src/data_sources/data_source.rs:641]: bump of qdrant client from 1.6 to 1.7

Comment (spolu) [reply core/src/data_sources/data_source.rs:641]: Introduces a new argument (shard_key) to their interface

Comment (fontanierh) [reply core/src/data_sources/data_source.rs:641]: got it thx !
